src/userdev/java/net/minecraftforge/gradle/userdev/DependencyManagementExtension.java

```diff
@@ -41,10 +57,96 @@ public Dependency deobf(Object dependency) {
+
+                final List<Node> dependenciesNodeList = (List<Node>) dependencies.stream()
+                        .filter(Node.class::isInstance)
+                        .map(Node.class::cast)
+                        .collect(Collectors.toList());
+
+                dependenciesNodeList.stream()
+                        .filter(el -> hasChildWithText(el, MAVEN_POM_NAMESPACE + "artifactId", "forge") && hasChildWithText(el, MAVEN_POM_NAMESPACE + "groupId", "net.minecraftforge"))
+                        .forEach(el -> el.parent().remove(el));
+
+
+                dependenciesNodeList.stream()
+                        .filter(el -> hasChildWithContainedText(el, MAVEN_POM_NAMESPACE + "version", "_mapped_"))
+                        .forEach(el -> setChildText(el, MAVEN_POM_NAMESPACE + "version", getVersionFrom(getChildText(el, MAVEN_POM_NAMESPACE + "version"))));
+            });
+        });
+
+        return mavenPublication;
+    }
+
+    @SuppressWarnings("unchecked")
+    private boolean hasChildWithText(final Node node, final String childKey, final String expectedValue) {
+        final NodeList children = node.getAt(QName.valueOf(childKey));
+        final List<Node> childList = (List<Node>) (children.stream()
+                .map(Node.class::cast)
+                .collect(Collectors.toList()));
+
+        return childList.stream()
+                .anyMatch(el -> el.text().equals(expectedValue));
+    }
+
+    @SuppressWarnings("unchecked")
+    private boolean hasChildWithContainedText(final Node node, final String childKey, final String expectedValue) {
+        final NodeList children = node.getAt(QName.valueOf(childKey));
+        final List<Node> childList = (List<Node>) (children.stream()
+                .map(Node.class::cast)
+                .collect(Collectors.toList()));
+
+        return childList.stream()
+                .anyMatch(el -> el.text().contains(expectedValue));
+    }
+
+    @SuppressWarnings("unchecked")
+    private String getChildText(final Node node, final String childKey) {
+        final NodeList children = node.getAt(QName.valueOf(childKey));
+        final List<Node> childList = (List<Node>) (children.stream()
+                .map(Node.class::cast)
+                .collect(Collectors.toList()));
+
+        return childList.stream().map(Node::text).findFirst().orElse("");
+    }
+
+    @SuppressWarnings("unchecked")
+    private void setChildText(final Node node, final String childKey, final String expectedValue) {
+        final NodeList children = node.getAt(QName.valueOf(childKey));
+        final List<Node> childList = (List<Node>) (children.stream()
+                .map(Node.class::cast)
+                .collect(Collectors.toList()));
+
+        childList.forEach(el -> el.setValue(expectedValue));
+    }
+
+    private String getVersionFrom(final String version) {
+        if (version.contains("_mapped_")) {
+            return version.split("_mapped_")[0];
```

Comment:
Is this what other code spots use? To me it would make more sense to substring.

Reply:
Yeah it uses the same code, I wondered as well. But I just copied it.

Reply:
Or more or less adapted it
The deobfuscating repo uses this technique to get the version and other information, which is the only place where it effectively does this, hence it being identical.